cmds/core/timeout/main.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2012-2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Run a command and kill it if it runs more than a specified duration
+//
+// Synopsis:
+//	timeout [-v] [-t duration-string] command [args...]
+//
+// Description:
+//	timeout will run the command until it succeeds or too much time has passed.
+//	The default timeout is 30s.
+//	If -v is set, it will show what it is running.
+//	If no args are given, it will just return.
+//
+// Example:
+//	$ timeout echo hi
+//	hi
+//	$
+//	$./timeout -v -t 5s bash -c 'sleep 40'
+//	$ 2022/03/31 14:47:27 Run ["bash" "-c" "sleep 40"]
+//	$ 2022/03/31 14:47:32 signal: killed
+//	$./timeout  -t 5s bash -c 'sleep 40'
+//	$ 2022/03/31 14:47:40 signal: killed
+//	$./timeout  -t 5s bash -c 'sleep 1'
+//	$
+
+//go:build !test
+// +build !test
+
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"log"
+	"os"
+	"os/exec"
+	"time"
+)
+
+var (
+	timeout   = flag.String("t", "30s", "Timeout for command")
+	verbose   = flag.Bool("v", false, "Log each attempt to run the command")
+	v         = func(string, ...interface{}) {}
+	ErrNoArgs = errors.New("Need at least a command to run")
+)
+
+func main() {
+	flag.Parse()
+	if *verbose {
+		v = log.Printf
+	}
+	if err := run(*timeout, v, flag.Args()...); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run(timeout string, v func(string, ...interface{}), args ...string) error {
+	if len(args) == 0 {
+		return ErrNoArgs
+	}
+	v("Run %q", args)
+	ctx := context.Background()
+	d, err := time.ParseDuration(timeout)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
In general I think log.Fatal* & panic should be reserved for main() ... it might be unexpected for a caller of a function for the function to panic. Granted this is a very small and self contained file - but I think it might make sense to return something like `return fmt.Errorf("failed to parse duration: %v", err)` as the caller on line 56 will handle the log.Fatal call